zookeeper/src/main/java/site/ycsb/db/zookeeper/ZKClient.java

```diff
@@ -0,0 +1,213 @@
+      watcher = null;
+    } else if ("true".equals(watchFlagString)) {
+      watcher = new SimpleWatcher();
+    } else {
+      throw new IllegalArgumentException("zookeeper.watchFlag only has two options:true or false");
+    }
+
+    long sessionTimeout;
+    String sessionTimeoutString = props.getProperty(SESSION_TIMEOUT_PROPERTY);
+    if (sessionTimeoutString != null) {
+      sessionTimeout = Integer.parseInt(sessionTimeoutString);
+    } else {
+      sessionTimeout = DEFAULT_SESSION_TIMEOUT;
+    }
+
+    try {
+      zk = new ZooKeeper(connectString, (int) sessionTimeout, new SimpleWatcher());
+    } catch (IOException e) {
+      throw new DBException("Creating connection failed.");
+    }
+  }
+
+  public void cleanup() throws DBException {
+    try {
+      zk.close();
+    } catch (InterruptedException e) {
+      throw new DBException("Closing connection failed.");
+    }
+  }
+
+  @Override
+  public Status read(String table, String key, Set<String> fields,
+                     Map<String, ByteIterator> result) {
+    String path = getPath(key);
+    try {
+      byte[] data = zk.getData(path, watcher, null);
+      if (data == null || data.length == 0) {
+        return Status.OK;
+      }
+      JSONObject jsonObject = (JSONObject)JSONValue.parse(new String(data));
+      Iterator<String> iterator = jsonObject.keySet().iterator();
+      while(iterator.hasNext()) {
+        String field = iterator.next();
+        String value = jsonObject.get(field).toString();
+        result.put(field, new StringByteIterator(value));
+      }
+      return Status.OK;
+    } catch (KeeperException | InterruptedException e) {
+      LOG.error("Error when reading a path:{},tableName:{}", path, table, e);
+      return Status.ERROR;
+    }
+  }
+
+  @Override
+  public Status insert(String table, String key,
+                       Map<String, ByteIterator> values) {
+    String path = getPath(key);
+    String data = getJsonStrFromByteMap(values);
+    try {
+      zk.create(path, data.getBytes(UTF_8), ZooDefs.Ids.OPEN_ACL_UNSAFE,
+          CreateMode.PERSISTENT);
+      return Status.OK;
+    } catch (KeeperException.NodeExistsException e1) {
+      return Status.OK;
+    } catch (KeeperException | InterruptedException e2) {
+      LOG.error("Error when inserting a path:{},tableName:{}", path, table, e2);
+      return Status.ERROR;
+    }
+  }
+
+  @Override
+  public Status delete(String table, String key) {
+
+    String path = getPath(key);
+    try {
+      zk.delete(path, -1);
+      return Status.OK;
+    } catch (InterruptedException | KeeperException e) {
+      LOG.error("Error when deleting a path:{},tableName:{}", path, table, e);
+      return Status.ERROR;
+    }
+  }
+
+  @Override
+  public Status update(String table, String key,
+                       Map<String, ByteIterator> values) {
+    String path = getPath(key);
+    String data = getJsonStrFromByteMap(values);
```

Comment:
This will effectively delete any fields not updated, right?

Reply:
- It will pick up one field from the multi-fields map, zookeeper supports the simple k-v data struct which the value is plaintext without any complex data struct, so only pick up one field is enough for benchmark test, to avoid the cost of reading and deserializing the current record.For example:

```
update **/user4177720500626811450** from
**{"field8":">A-8W1#5z&)n2O5$6h)0.(<|' <=\"x3Dy>Qq,N5)\/~>\"`*3z 3*0Q%:!j6Dg2^? #>.92$:\"-W}:T}5>`?-j3N1%!b0?,(I%7$,4"}**
to
**{"field5":".Z9)Ew:\\q:&x3@i+I;%V}.<0#,p61 94 0Cc'Zs?&,=>$9Ri:=:%M14I+59b:=6$Tq4;4$A132(1Lk94`63v\/\\u'C=5Jy#\\c37x'"}**
```

- the UT was added.